internal/token/jwt.go

```diff
@@ -0,0 +1,78 @@
+package token
+
+import (
+	"errors"
+	"github.com/benbjohnson/clock"
+	"github.com/golang-jwt/jwt/v5"
+	"kg/procurement/cmd/config"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type jwtManager struct {
+	cfg   config.Token
+	clock clock.Clock
+}
+
+func (s *jwtManager) GenerateToken(spec ClaimSpec) (string, error) {
+	if s.cfg.Secret == "" {
+		return "", errors.New("secret key is empty")
+	}
+
+	tokenObject := jwt.NewWithClaims(jwt.SigningMethodHS256, Claims{
+		RegisteredClaims: jwt.RegisteredClaims{
+			Subject:   spec.UserID,
+			ExpiresAt: jwt.NewNumericDate(s.clock.Now().UTC().Add(30 * 24 * time.Hour)), // one month from now
+			IssuedAt:  jwt.NewNumericDate(s.clock.Now().UTC()),
+			ID:        uuid.NewString(),
```

Comment:
Could you explain to me why you put an ID inside JWT token? It is actually my first time seeing this case

Reply:
ID is "jti" in JWT. It can be used to prevent the JWT from being replayed. [This is](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.7) a great read for you!